app/assets/javascripts/startpage/startPageServices.coffee

```diff
@@ -66,5 +67,34 @@ startPageServices.service("StartPageStateData", () ->
     this.getResponsesForTopic = (topicId) ->
         if topicId of responsesForSelectedTopics then responsesForSelectedTopics[topicId] else {}
 
+    # Progress bar interface
+    progressBar = {
+        latestTopic: 0
+        topicsCompleted: 0
+    }
+
+    # Progress bar state
+    Object.defineProperty(this, "progressBar", {
+        get: () -> progressBar
+        })
+    this.getNumQuestions = -> progressBar.numQuestions
```

Comment:
you don't need the getters. You just simply do service.progressBar.variable.

Setters are necessary, though.